### PR TITLE
expression: fix pb2expr panic.

### PR DESCRIPTION
**Pull request by hanfei1991**
every time we get builtinFunctionSignature, we should copy it.
A temp fix is to get function class directly.
@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: expression/distsql_builtin_test.go [hunk 1]
```diff
@@ -164,6 +164,10 @@ func (s *testEvalSuite) TestEval(c *C) {
 			types.NewIntDatum(0),
 		},
 		{
+			buildExpr(tipb.ExprType_And, buildExpr(tipb.ExprType_And, types.NewIntDatum(1), types.NewIntDatum(1)), buildExpr(tipb.ExprType_And, types.NewIntDatum(0), types.NewIntDatum(1))),
```
Comment (shenli) [inline expression/distsql_builtin_test.go]: This case will meet panic without this PR?

Comment (hanfei1991) [inline expression/distsql_builtin_test.go]: yes

**Comment by shenli** [comment]
LGTM
@coocood @zimulala PTAL

**Comment by coocood** [comment]
LGTM

File: expression/distsql_builtin.go [hunk 2]
```diff
@@ -17,70 +17,69 @@ import (
...
-	f, ok := distFuncs[exprType]
+func newDistSQLFunction(sc *variable.StatementContext, exprType tipb.ExprType, args []Expression) (Expression, error) {
+	name, ok := distFuncs[exprType]
 	if !ok {
 		return nil, errFunctionNotExists.GenByArgs(exprType)
 	}
 	// TODO: Too ugly...
 	ctx := mock.NewContext()
 	ctx.GetSessionVars().StmtCtx = sc
-	f.init(args, ctx)
```
Comment (zimulala) [inline expression/distsql_builtin.go]: Remove the function of `init`.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**